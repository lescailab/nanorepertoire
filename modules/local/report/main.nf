process REPORT {
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/bioconductor-biostrings_bioconductor-genomicfeatures_bioconductor-ggtree_bioconductor-gviz_pruned:d368c94e0967e319' :
        'community.wave.seqera.io/library/bioconductor-biostrings_bioconductor-genomicfeatures_bioconductor-ggtree_bioconductor-gviz_pruned:fdfe24a734a90d8f' }"

    input:
    path report
    path loop_tree
    path clustersummaries
    path cdr3histograms
    path cdr3tables
    val  metadata

    output:
    path '*.RData', emit: rdata
    path '*.html', emit: report
    path 'cdr3_boost_overview_table.tsv', emit: results
    path 'sampledata.tsv', emit: sampledata

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def clusterList = clustersummaries.join(',')
    def histoList = cdr3histograms.join(',')
    def tableList = cdr3tables.join(',')

    """
    # Create sampledata.tsv within the task execution context
    echo "ID\tindividual\timmunisation\tboost" > sampledata.tsv
    ${metadata.collect { map ->
        "echo '${map.sampleID}\t${map.individualID}\t${map.immunisation}\t${map.boost}' >> sampledata.tsv"
    }.join('\n')}

    ln -s '$moduleDir/nibsc_report.css' .

    quarto render analysis_report.qmd \\
        -P clusterList:'${clusterList}' \\
        -P histoList:'${histoList}' \\
        -P tableList:'${tableList}' \\
        -P sampleData:'sampledata.tsv' \\
        -P sizeThreshold:'${params.cluster_size_threshold}' \\
        -P loopFile:'loop_tree.qmd' \\
        -P calcTree:'${params.calculate_tree}' \\
        --output '${report.baseName}.html'
    """
}
