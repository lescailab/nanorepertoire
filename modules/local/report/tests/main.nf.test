nextflow_process {

    name "Test Process REPORT"
    script "../main.nf"
    process "REPORT"
    config "./nextflow.config"
    tag "modules"
    tag "modules_nfcore"
    tag "report"

    test("reporting") {

        when {
            process {
                """
                // input_ch
                input[0] = [
                    file("$projectDir/assets/analysis_report.qmd")
                ]
                input[1] = [
                    file("$projectDir/assets/loop_tree.qmd")
                ]
                input[2] = [
                    file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_clusters.summary', checkIfExists: true)
                ]
                input[3] = [
                    file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_cdr3.hist', checkIfExists: true)
                ]
                input[4] = [
                    file('https://raw.githubusercontent.com/DavideBag/testdata/dev/nanobodies/report/test_cdr3.tsv', checkIfExists: true)
                ]
                input[5] = [
                    [id: 'test', sampleID: 'sample1', individualID: 'ind1', immunisation: 'imm1', boost: 'boost1']
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.stdout).match()}
            )
        }
    }
}
