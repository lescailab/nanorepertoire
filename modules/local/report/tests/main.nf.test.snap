{
    "reporting": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-14T09:48:03.722276"
    }
}